src/withAutofillProps.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react'
+import wrapDisplayName from 'recompose/wrapDisplayName'
+
+const startAnimationName = 'onAutofillStart'
+const endAnimationName = 'onAutofillCancel'
+const styleElementId = 'autofill-style'
+
+const isWebkit = () => navigator.userAgent.indexOf('WebKit') !== -1
+
+function emptyAnimation(animationName) {
+  return `
+    @keyframes ${animationName} {
+      from {/**/}
+      to {/**/}
+    }
+  `
+}
+
+function injectStyle(style) {
+  let styleElement = document.getElementById(styleElementId)
+
+  if (!styleElement) {
+    styleElement = document.createElement('style')
+    styleElement.id = styleElementId
+    document.head.appendChild(styleElement)
+  }
+
+  let styleSheet = styleElement.sheet
+
+  styleSheet.insertRule(style, styleSheet.cssRules.length)
+}
+
+function injectKeyFrames() {
+  const autofillStartEmptyKeyframesStyle = emptyAnimation(startAnimationName)
+  const autofillEndEmptyKeyframesStyle = emptyAnimation(endAnimationName)
+
+  injectStyle(autofillStartEmptyKeyframesStyle)
+  injectStyle(autofillEndEmptyKeyframesStyle)
+}
+
+function injectAutofillHook() {
+  const autofillHook = `
+    input:-webkit-autofill {
+      // Expose a hook for JavaScript when autofill is shown.
+      // JavaScript can capture 'animationstart' events
+      animation-name: ${startAnimationName};
+
+      // Make the backgound color become yellow _really slowly_
+      transition: background-color 50000s ease-in-out 0s;
+    }
+  `
+
+  const notAutofillHook = `
+    input:not(:-webkit-autofill) {
+      // Expose a hook for JavaScript when autofill is no longer shown.
+      // JavaScript can capture 'animationstart' events
+      animation-name: ${endAnimationName};
+    }`
+
+  injectStyle(autofillHook)
+  injectStyle(notAutofillHook)
+}
+
+function registerAutofill() {
```

Comment:
Yeah... if you have 5 components wrapped with this HOC then you can't just remove `<style>` from DOM when one of them is about to unmount. But from another hand it's a dirty side effect which should be cleaned up somehow if nobody is using it anymore.